crates/bevy_pbr/src/light_probe/environment_filter.wgsl

```diff
@@ -0,0 +1,396 @@
+        var xi = hammersley_2d(i, sample_count);
+        xi = fract(xi + vector_noise.rg); // Apply Cranley-Patterson rotation
+        
+        // Sample the GGX distribution with the spherical-cap VNDF method
+        let light_dir = sample_visible_ggx(xi, roughness, normal, view);
+        
+        // Calculate weight (N·L)
+        let NoL = dot(normal, light_dir);
+        
+        if (NoL > 0.0) {
+            // Reconstruct the microfacet half-vector from view and light and compute PDF terms
+            let half_vector = normalize(view + light_dir);
+            let NoH = dot(normal, half_vector);
+            let VoH = dot(view, half_vector);
+            let NoV = dot(normal, view);
+            
+            // Get the geometric shadowing term
+            let G = G_Smith(NoV, NoL, roughness);
+            
+            // PDF that matches the bounded-VNDF sampling
+            let pdf = ggx_vndf_pdf(view, NoH, roughness);
+            
+            // Calculate LOD using filtered importance sampling
+            // This is crucial to avoid fireflies and improve quality
+            let width = f32(size.x);
+            let lod = calculate_environment_map_lod(pdf, width, f32(sample_count));
+            
+            // Get source mip level - ensure we don't go negative
+            let source_mip = max(0.0, lod);
+            
+            // Sample environment map with the light direction
+            var sample_color = sample_environment(light_dir, source_mip).rgb;
+            sample_color = tonemap(sample_color);
+            
+            // Accumulate weighted sample, including geometric term
+            radiance += sample_color * NoL * G;
+            total_weight += NoL * G;
+        }
+    }
+    
+    // Normalize by total weight
+    if (total_weight > 0.0) {
+        radiance = radiance / total_weight;
+    }
+
+    // Reverse tonemap
+    radiance = reverse_tonemap(radiance);
+    
+    // Write result to output texture
+    textureStore(output_texture, coords, face, vec4f(radiance, 1.0));
+}
+
+// Calculate spherical coordinates using spiral pattern
+// and golden angle to get a uniform distribution
+fn uniform_sample_sphere(i: u32, normal: vec3f) -> vec3f {
+    // Get stratified sample index
+    let index = i % constants.sample_count;
+    
+    let golden_angle = 2.4;
+    let full_sphere = f32(constants.sample_count) * 2.0;
+    let z = 1.0 - (2.0 * f32(index) + 1.0) / full_sphere;
+    let r = sqrt(1.0 - z * z);
+    
+    let phi = f32(index) * golden_angle;
+    
+    // Create the direction vector
+    let dir_uniform = vec3f(
+        r * cos(phi),
+        r * sin(phi),
+        z
+    );
+
+    let tangent_frame = build_orthonormal_basis(normal);
+    return normalize(tangent_frame * dir_uniform);
+}
+
+// from bevy_solari/src/scene/sampling.wgsl
+fn sample_cosine_hemisphere(normal: vec3<f32>, rng: ptr<function, u32>) -> vec3<f32> {
```

Comment:
Same here, seems a shame to copy paste this. Imo we could probably extract sample_uniform_hemisphere, sample_cosine_hemisphere, and the VNDF functions into a new file, and keep solari's sampling.wgsl just for solari lighting, as that's what the rest of the file does in practice anyways.